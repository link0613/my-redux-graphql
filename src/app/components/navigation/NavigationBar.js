// @flow

// #region imports
import React, { type SyntheticEvent } from 'react';
import PropTypes from 'prop-types';
import Humburger from './humburger/Humburger';
import LeftNav from './leftNav/LeftNav';
import RightNav from './rightNav/RightNav';
// #endregion

// #region flow types
export type Props = {
  brand?: string,
  userIsAuthenticated?: boolean,
  handleLeftNavItemClick: (event: SyntheticEvent<>, viewName: string) => any,
  handleRightNavItemClick: (event: SyntheticEvent<>, viewName: string) => any,
  navModel: {
    leftLinks: Array<{
      label: string,
      link: string,
    }>,
    rightLinks: Array<{
      label: string,
      link: string,
    }>,
  },
};
// #endregion

const NavigationBar = ({
  brand,
  navModel,
  handleLeftNavItemClick,
  handleRightNavItemClick,
  userIsAuthenticated,
}: Props) => {
  return (
    <nav className="navbar navbar-default">
      <div className="containersCustom">
        <div className="navbar-header">
          {<Humburger />}
          <a className="navbar-brand">{brand}</a>
        </div>
        <div
          className="collapse navbar-collapse"
          id="bs-example-navbar-collapse-1"
        >
          <ul className="nav navbar-nav">
            {
              <LeftNav
                leftLinks={navModel.leftLinks}
                onLeftNavButtonClick={handleLeftNavItemClick}
                userIsAuthenticated={userIsAuthenticated}
              />
            }
          </ul>
          <ul className="nav navbar-nav navbar-right">
            {
              <RightNav
                rightLinks={navModel.rightLinks}
                onRightNavButtonClick={handleRightNavItemClick}
                userIsAuthenticated={userIsAuthenticated}
              />
            }
          </ul>
        </div>
      </div>
    </nav>
  );
};

// #region static props
NavigationBar.propTypes = {
  brand: PropTypes.string,
  userIsAuthenticated: PropTypes.bool.isRequired,
  handleLeftNavItemClick: PropTypes.func,
  handleRightNavItemClick: PropTypes.func,
  navModel: PropTypes.shape({
    leftLinks: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        link: PropTypes.string.isRequired,
      }),
    ).isRequired,
    rightLinks: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        link: PropTypes.string.isRequired,
      }),
    ).isRequired,
  }),
};

NavigationBar.defaultProps = {
  brand: 'brand',
};

NavigationBar.displayName = 'NavigationBar';
// #endregion

export default NavigationBar;
