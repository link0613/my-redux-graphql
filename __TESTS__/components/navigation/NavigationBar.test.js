import React from 'react';
import NavigationBar from '../../../src/app/components/navigation/NavigationBar';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';

describe('NavigationBar (navigation) component', () => {
  it('renders as expected (not authenticated)', () => {
    const props = {
      brand: 'brand',
      handleLeftNavItemClick: () => {},
      handleRightNavItemClick: () => {},
      userIsAuthenticated: false,
      navModel: {
        rightLinks: [
          {
            link: 'link',
            label: 'label',
            viewName: '/',
            btnLink: { showWhenUserAuth: false, alwaysShows: true },
          },
          {
            link: 'link',
            label: 'label',
            viewName: '/',
            btnLink: { showWhenUserAuth: true, alwaysShows: false },
          },
          {
            link: 'link',
            label: 'label',
            viewName: '/',
            btnLink: { showWhenUserAuth: true, alwaysShows: true },
          },
          {
            link: 'link',
            label: 'label',
            viewName: '/',
            btnLink: { showWhenUserAuth: false, alwaysShows: false },
          },
        ],
        leftLinks: [
          {
            link: 'link',
            label: 'label',
            viewName: '/',
            btnLink: { showWhenUserAuth: false, alwaysShows: true },
          },
          {
            link: 'link',
            label: 'label',
            viewName: '/',
            btnLink: { showWhenUserAuth: true, alwaysShows: false },
          },
          {
            link: 'link',
            label: 'label',
            viewName: '/',
            btnLink: { showWhenUserAuth: true, alwaysShows: true },
          },
          {
            link: 'link',
            label: 'label',
            viewName: '/',
            btnLink: { showWhenUserAuth: false, alwaysShows: false },
          },
        ],
      },
    };
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <NavigationBar {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
