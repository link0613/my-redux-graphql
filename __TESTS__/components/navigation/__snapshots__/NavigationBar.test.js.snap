// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar (navigation) component renders as expected (not authenticated) 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="containersCustom"
    >
      <div
        className="navbar-header"
      >
        <button
          className="navbar-toggle collapsed"
          data-target="#bs-example-navbar-collapse-1"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
        <a
          className="navbar-brand"
        >
          brand
        </a>
      </div>
      <div
        className="collapse navbar-collapse"
        id="bs-example-navbar-collapse-1"
      >
        <ul
          className="nav navbar-nav"
        >
          <ul
            className="nav navbar-nav"
          >
            <li>
              <a
                href="link"
                onClick={[Function]}
              >
                label
              </a>
            </li>
            <li>
              <a
                href="link"
                onClick={[Function]}
              >
                label
              </a>
            </li>
            <li>
              <a
                href="link"
                onClick={[Function]}
              >
                label
              </a>
            </li>
            <li>
              <a
                href="link"
                onClick={[Function]}
              >
                label
              </a>
            </li>
          </ul>
        </ul>
        <ul
          className="nav navbar-nav navbar-right"
        >
          <ul
            className="nav navbar-nav navbar-right"
          />
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
